vllm/core/block/cpu_gpu_block_allocator.py

```diff
@@ -105,11 +105,24 @@ def __init__(
             Device.GPU: gpu_block_allocator,
         }
 
+        self._null_block: Optional[Block] = None
+
         self._block_ids_to_allocator: Dict[int, BlockAllocator] = {}
         for _, allocator in self._allocators.items():
             for block_id in allocator.all_block_ids:
                 self._block_ids_to_allocator[block_id] = allocator
 
+    @property
+    def null_block(self) -> Block:
+        if self._null_block is None:
+            self._null_block = self.allocate_mutable(None, Device.GPU)
+
+            def fail(token_ids: List[int]):
+                raise ValueError("null_block should not be modified")
+
+            self._null_block.append_token_ids = fail  # type: ignore
```

Comment:
this is hacky, can we do this in a cleaner way?

Reply:
e.g. a new special block implementation that defers to the internal one except for `append_token_ids`